plugin/src/main/java/org/jenkinsci/plugins/structs/describable/ParameterType.java

```diff
@@ -61,47 +61,47 @@ static ParameterType of(Type type) {
                 if (c.isArray()) {
                     return new ArrayType(c);
                 }
-                // Assume it is a nested object of some sort.
-                Set<Class<?>> subtypes = DescribableModel.findSubtypes(c);
-                if ((subtypes.isEmpty() && !Modifier.isAbstract(c.getModifiers())) || subtypes.equals(Collections.singleton(c))) {
-                    // Probably homogeneous. (Might be concrete but subclassable.)
-                    return new HomogeneousObjectType(c);
-                } else {
-                    // Definitely heterogeneous.
-                    Map<String,List<Class<?>>> subtypesBySimpleName = new HashMap<String,List<Class<?>>>();
-                    for (Class<?> subtype : subtypes) {
-                        String simpleName = subtype.getSimpleName();
-                        List<Class<?>> bySimpleName = subtypesBySimpleName.get(simpleName);
-                        if (bySimpleName == null) {
-                            subtypesBySimpleName.put(simpleName, bySimpleName = new ArrayList<Class<?>>());
-                        }
-                        bySimpleName.add(subtype);
+            }
+            if (Types.isSubClassOf(type, Collection.class)) {
+                return new ArrayType(type, of(Types.getTypeArgument(Types.getBaseClass(type,Collection.class), 0, Object.class)));
+            }
+            Class<?> c = Types.erasure(type);
+            // Assume it is a nested object of some sort.
```

Comment:
Most of this is actually reindentation, making the diff look a lot bigger than it actually is. Had to apply the WS ignore setting in GitHub to see it.